tiledb/sm/enums/datatype.h

```diff
@@ -51,7 +51,12 @@ enum class Datatype : uint8_t {
 #undef TILEDB_DATATYPE_ENUM
 };
 
-/** Returns the datatype size. */
+/**
+ *  Returns the datatype size.
+ *
+ *  This function also serves as a datatype validation function;
+ *  If 0 is returned, the input Datatype is invalid.
+ */
 inline uint64_t datatype_size(Datatype type) noexcept {
```

Comment:
I know this a pre-existing defect, but `uint64_t` is not right here. It should be `size_t`. I don't know how much trouble it would be to change here, but this seems like a good PR to combine it with if the changes aren't extensive.

Reply:
Potentially out of scope for this PR; will revisit after next push